doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+        - providing a IPC
+        - directly modifying the memory)
+(Why: To enable a fast modify-update-test cycle during tunning.)
+
+Tunning SHOULD be possible from the PF native API.
+Why: In order to let the host system offer implement its own tunning mechanism.)
+
+Users MAY be able to modify the parameters (types, identifiers, tree...) with minimal effort (see previous requirement).
+(Note: The reference implementation does not support it.)
+(Why: To enable a fast modify-update-test cycle on PF configuration.)
+
+Users SHOULD be able to modify the parameter values at any time.
+This change SHOULD not be overwritten without a user action.
+(Note: user overwritten user action could be a log out, leaving some tunning mode, forcing an inference engine update...)
+(Why: Even if a parameter is managed by the inference engine,
+      it often is useful (test, debugging) to overwrite its value temporally.).
+
+A PF tunning capability MAY be disabled in a context where no tunning is needed.
+(Why: The reference implementation does so (phone end users can not change the tunning).)
+
+# Command line interface
+
+The PF MAY offer a command line interface that binds to it's IPC.
+(Why: to have a reference way to interact with a PF without implementing it's IPC protocol.)
+
+This command line interface SHOULD support all tunning and introspection ability.
+(Why: In order to be used in scripting and live tunning/debugging on an embedded system.)
+
+This command line interface MAY offer argument auto completion.
+(Why: Is more user friendly.)
+
+# Bindings
+The PF SHOULD expose it's api in C.
+(Why: The PF aims to be a hardware abstraction thus middle ware which is often written in C or a language compatible with C.
+      Virtually all programing language support C Foreign Procedure Call,
+      having a C api ease integration whichever the host language is.)
+
+The PF MAY expose it's api to multiple programing language.
+(Why: The reference implementation has python bindings.)
+
+# Next
+The following requirements are note implemented in the reference implementation and are
+to be consider draft.
+
+PF MAY support at least:
+    - window
+    - linux
+    - ios
+(Why: As the reference PF implementation leave it's original android environement,
+      needs emerge to use it on other platform.)
+
+The PF host API SHOULD be structured.
+IE: the FW when requested for a list of domains should return a list of structured
+object, each containing configuration objects, containing their values.
+(Why: The reference implementation has an string oriented API.
+      Eg: The list of domains is returned as a concatenation of domains name in one big string.
+      This leads to hard an hard to use api from C and C++ code.)
+
+The PF host API SHOULD expose parameter values with the same API syncer use.
+(Why: The current reference implementation abstracts the memory layout of parameters.
+      This memory layout is specified in the parameter structure thus TODO)
```

Comment:
Need to add all performance related requirements. If any. If no pKPI then explicitely write there is no KPI...

Wht about requirements within the XML and tags "API"?